### PR TITLE
Handles situations where strings are passed when lists are expected

### PR DESCRIPTION
**Pull request by mratford**

### CODE REVIEW

File: iam_builder/iam_schema.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    "type": "array",
+                    "items": {"type": "string"},
+                },
+            },
+        },
+    },
+}
+
+
+def validate_iam(config):
```
Comment (s-ducks) [inline iam_builder/iam_schema.py:60]:
```suggestion
def validate_iam(config: dict):
```

File: iam_builder/iam_schema.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import jsonschema
+
+IAM_SCHEMA = {
```
Comment (s-ducks) [inline iam_builder/iam_schema.py:3]: Could you make this a json file and load it in please? [the same way data linter does](https://github.com/moj-analytical-services/data_linter/blob/main/data_linter/constants.py).
You should just be able to use `json.dump` to dump this dictionary to a json file and save yourself the work

**Comment by s-ducks** [comment]
The validation doesn't throw an error when the incorrect field name is put in, for example a common mistake would be to put `read` instead of `read_only`, this is allowed according to the schema but produces the incorrect output. I think this could be solved by adding `additionalProperties: false`

File: iam_builder/iam_schema.py (hunk 2 above)
Comment (mratford) [reply iam_builder/iam_schema.py]: Done - cbff1e4

**Comment by mratford** [comment]
> The validation doesn't throw an error when the incorrect field name is put in, for example a common mistake would be to put `read` instead of `read_only`, this is allowed according to the schema but produces the incorrect output. I think this could be solved by adding `additionalProperties: false`

df077c7

File: iam_builder/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+class IAMValidationError(Exception):
```
Comment (isichei) [inline iam_builder/exceptions.py:1]: Just wondering if we should be this inherit from the jsonschema ValidationError class? That way any usage of IAM builder where someone does a try except on a ValidationError will still be caught?

**Review by isichei — APPROVED** [review]
See comment to see what you think. Everything else looks good

File: iam_builder/exceptions.py (hunk 3 above)
Comment (mratford) [reply iam_builder/exceptions.py:1]: That's a fair point, I did originally have it like that but changed it to make it simpler.

**Pull request merged**